jsf-showcase-webapp/src/test/java/com/liferay/faces/test/showcase/facesrequestcontext/FacesRequestContextGeneralTester.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2000-2016 Liferay, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.liferay.faces.test.showcase.facesrequestcontext;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import com.liferay.faces.test.selenium.Browser;
+import com.liferay.faces.test.selenium.assertion.SeleniumAssert;
+import com.liferay.faces.test.showcase.buttonlink.ButtonLinkTester;
+
+
+/**
+ * @author  Kyle Stiemann
+ * @author  Philip White
+ */
+public class FacesRequestContextGeneralTester extends ButtonLinkTester {
+
+	@Test
+	public void runFacesRequestContextGeneralTest() throws Exception {
+
+		Browser browser = Browser.getInstance();
+		navigateToUseCase(browser,"util", "facesrequestcontext", "general");
+
+		// Test facesrequestcontext modal is visible.
+		browser.click("(//*[contains(@value,'Show Modal')])[1]");
+		browser.waitUntil(ExpectedConditions.visibilityOfElementLocated(By.xpath(submitButton1Xpath)));
```

Comment:
`browser.waitForElementVisible()` can be used here to make this line more clear. Please use `browser.wait*()` methods in place of the longer form `browser.waitUntil()` method where possible.